### PR TITLE
iOS: add `KeyValueStore` protocol conformance to `UserDefaults`

### PR DESCRIPTION
**Pull request by jpsim**
Continued from #2441.

Description: Parallels Android implementation based on SharedPreferences.
Risk Level: Low
Testing: Application

Co-authored-by: Mike Schore <mike.schore@gmail.com>
Signed-off-by: JP Simard <jp@jpsim.com>

### CODE REVIEW

File: library/swift/BUILD [hunk 1]
```diff
@@ -9,45 +9,7 @@ licenses(["notice"])  # Apache 2
...
-        "StreamClientImpl.swift",
-        "StreamIntel.swift",
-        "StreamPrototype.swift",
-        "TestEngineBuilder.swift",
-        "Trailers.swift",
-        "UpstreamHttpProtocol.swift",
-        "filters/*.swift",
-        "grpc/*.swift",
-        "mocks/*.swift",
-        "stats/*.swift",
```
Comment (goaway) [inline library/swift/BUILD:50]: Personally, I'd prefer we explicitly enumerate source files in build targets and avoid blobs. I feel it obscures information about what's being pulled in, and it has the potential to lead to surprises when files are moved around.

Upstream Envoy tends to follow the pattern of explicit enumeration of source files as well, even when that list is large.

**Review by goaway — CHANGES_REQUESTED** [review]
Thanks @jpsim, this is a nifty approach! Left one comment, but otherwise looks good.

File: library/swift/BUILD (hunk 1 above)
Comment (jpsim) [reply library/swift/BUILD:50]: Done. I don't feel strongly but the previous approach was inconsistent with a mix of explicit files and globs.

As of 4d77493 everything is being explicitly set.

**Review by Augustyniak — APPROVED** [review]

**Pull request merged**